src/Symfony/Component/Form/Extension/Csrf/EventListener/CsrfValidationListener.php

```diff
@@ -89,13 +95,15 @@ public function __construct($fieldName, $tokenManager, $tokenId, $errorMessage,
         $this->errorMessage = $errorMessage;
         $this->translator = $translator;
         $this->translationDomain = $translationDomain;
+        $this->serverParams = $serverParams ?: new ServerParams();
     }
 
     public function preSubmit(FormEvent $event)
     {
         $form = $event->getForm();
+        $postRequestSizeExceeded = $form->getConfig()->getMethod() === 'POST' && $this->serverParams->hasPostMaxSizeBeenExceeded();
```

Comment:
POST request size is not exclusive to forms, so adding too much logic into a Form component class doesn't make much sense to me. 

Would be better to move the logic around max POST (and PATCH, PUT) to HttpFoundation and then use that component here instead. 

Sent from my iPhone

> On 29 Jul 2016, at 14:02, Jules Pietri <notifications@github.com> wrote:
> 
> In src/Symfony/Component/Form/Extension/Csrf/EventListener/CsrfValidationListener.php:
> 
> >      }
> >  
> >      public function preSubmit(FormEvent $event)
> >      {
> >          $form = $event->getForm();
> > +        $postRequestSizeExceeded = $form->getConfig()->getMethod() === 'POST' && $this->serverParams->hasPostMaxSizeBeenExceeded();
> The logic is the same for the native request handler too (ref https://github.com/HeahDude/symfony/blob/master/src/Symfony/Component/Form/NativeRequestHandler.php#L68), so I suggest to keep things simple here
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>